### DISCUSSION TITLE
Is it possible to boot the same LTSP image with different kernels (for example 5.10 and 6.1 on Debian 11)?

### DISCUSSION THREAD

**Discussion by N0rbert** [ltsp/ltsp#906] [General] [answered]
I have configured LTSP server with 23.01 version on Debian 11 to serve 40 clients.
To support old and new hardware I plan to use the same single LTSP VM image with different kernels - 5.10 from *bullseye* (for 10 PCs) and 6.1 from *bullseye-backports* (for 30 PCs).
  
I have successfully installed both kernels into LTSP VM image, but I do not know how to correctly add them to iPXE boot menu without dumb hardcoding. 
Ideally I want to have the following solution:
1. provide and show the line with 5.10 kernel in iPXE boot menu only for 10 systems with known MAC addresses, auto boot it here;
1. show the line with 6.1 kernel and auto boot in on the 30 systems with other known MAC addresses.

Is it possible to apply my configuration using some  `ltsp` (*-kernel*, *-initrd*) commands?

**Comment by alkisg** [comment, marked as answer]
One way would be to use an [ltsp-en.ipxe template](https://github.com/ltsp/ltsp/wiki/Non-LTSP-iPXE-entries) to modify the ipxe menu however you want.

Another way would be to:

    ln -s x86_64.img /srv/ltsp/images/x86_64-alt.img

And to use `ltsp kernel --kernel-initrd="regex..." x86_64-alt` parameter to select the lower kernel for that one.
And to put `DEFAULT_IMAGE=x86_64-alt` in ltsp.conf under the `[MAC:ADDRESSES]` of the clients you want.
